h2o-core/src/test/java/water/ScopeTest.java

```diff
@@ -0,0 +1,56 @@
+package water;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests for Scope
+ */
+public class ScopeTest extends TestUtil {
+  @BeforeClass() public static void setup() { stall_till_cloudsize(1); }
+
+  @Test
+  public void testEnterAndExit() throws Exception {
+    Scope.enter();
+    Scope.exit(); // not supposed to leak anything
+  }
+
+  @Test
+  public void testEnterAndExit_with_something_to_keep() throws Exception {
+    Scope.enter();
+    Key k1 = Key.make("k11");
+    Key k2 = Key.make("k12");
+    Key k3 = Key.make("k13");
+    Scope.exit(k3, k1, k2); // not supposed to leak anything
```

Comment:
Leaks are checked AFTER class, not after each test.

If this particular tests starts leaking keys, it will be wiped out by the next test that calls removeKeysRegardless().